pcsx2/gui/MemoryCardFolder.cpp

```diff
@@ -450,17 +445,21 @@ bool FolderMemoryCard::AddFile( MemoryCardFileEntry* const dirEntry, const wxStr
 		}
 
 		newFileEntry->entry.data.length = filesize;
-		u32 fileDataStartingCluster = GetFreeDataCluster();
-		newFileEntry->entry.data.cluster = fileDataStartingCluster;
-
-		// mark the appropriate amount of clusters as used
-		u32 dataCluster = fileDataStartingCluster;
-		m_fat.data[0][0][dataCluster] = LastDataCluster | DataClusterInUseMask;
-		for ( unsigned int i = 0; i < countClusters - 1; ++i ) {
-			u32 newCluster = GetFreeDataCluster();
-			m_fat.data[0][0][dataCluster] = newCluster | DataClusterInUseMask;
-			m_fat.data[0][0][newCluster] = LastDataCluster | DataClusterInUseMask;
-			dataCluster = newCluster;
+		if ( filesize != 0 ) {
+			u32 fileDataStartingCluster = GetFreeDataCluster();
+			newFileEntry->entry.data.cluster = fileDataStartingCluster;
+
+			// mark the appropriate amount of clusters as used
+			u32 dataCluster = fileDataStartingCluster;
+			m_fat.data[0][0][dataCluster] = LastDataCluster | DataClusterInUseMask;
+			for ( unsigned int i = 0; i < countClusters - 1; ++i ) {
+				u32 newCluster = GetFreeDataCluster();
+				m_fat.data[0][0][dataCluster] = newCluster | DataClusterInUseMask;
+				m_fat.data[0][0][newCluster] = LastDataCluster | DataClusterInUseMask;
+				dataCluster = newCluster;
+			}
+		} else {
+			newFileEntry->entry.data.cluster = MemoryCardFileEntry::EmptyFileCluster;
```

Comment:
That's not my decision, that's what developers at Sony did. See here: http://www.csclub.uwaterloo.ca:11068/mymc/ps2mcfs.html

Specifically the `cluster` entry in the Directory Entry table.
> First cluster of the file, or `0xFFFFFFFF` for an empty file. In "." entries this the first cluster of this directory's parent directory instead. Relative to alloc_offset.

Also, length is set to 0 before the branch.